plugins/cilium-docker/driver/driver.go

```diff
@@ -299,37 +300,36 @@ func (driver *driver) createEndpoint(w http.ResponseWriter, r *http.Request) {
 		},
 	}
 
+	removeLinkOnErr := func(link netlink.Link) {
+		if err != nil && link != nil && !reflect.ValueOf(link).IsNil() {
+			if err = netlink.LinkDel(link); err != nil {
```

Comment:
IMO it's fine, because `removeLinkOnErr` is called after the main function has returned, so there is no need to preserve the global error. Or am I missing something?

Reply:
If `netlink.LinkDel` errors out, you are overwriting the initial "global" error, when returning to the caller of `createEndpoint`, the caller does not know why the `netlink.LinkDel` was called in the first place.

Reply:
Changed the code to not to use the global error to avoid any possible confusion in the future.